bin/l10n-sync.js

```diff
@@ -0,0 +1,307 @@
+    prevOutput = cp
+      .execFileSync(executable, args, { input: prevOutput })
+      .toString();
+  }
+  return prevOutput.toString();
+}
+
+/**
+ * Pause with a message and wait for the enter as a confirmation.
+ * The propt will not be displayed if the `-y` argument is given to the script.
+ * This is mainly used by the CircleCI automation.
+ *
+ * @param {string} message to be displayed before the prompt.
+ * @returns {Promise<void>}
+ */
+async function pauseWithMessageIfNecessary(message = '') {
+  if (SKIP_PROMPTS) {
+    return;
+  }
+
+  if (message.length > 0) {
+    message += '\n';
+  }
+  message += 'Press ENTER when youʼre ready...';
+
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  await promisify(rl.question).call(rl, message);
+  rl.close();
+}
+
+/**
+ * Check if Git workspace is clean.
+ *
+ * @returns {Promiose<void>}
+ * @throws Will throw an error if workspace is not clean.
+ */
+async function checkIfWorkspaceClean() {
+  console.log('>>> Checking if the workspace is clean for the operations.');
+  // git status --porcelain --ignore-submodules -unormal
+  const status = await logAndExec(
+    'git',
+    'status',
+    '--porcelain',
+    '--ignore-submodules',
+    '-unormal'
+  );
+
+  if (status.length) {
+    throw new Error(
+      'Your workspace is not clean. Please commit or stash your changes.'
+    );
+  }
+
+  console.log('Workspace is clean.');
+}
+
+/**
+ * Finds the Git upstream remote and returns it.
+ *
+ * @returns {string} Name of the upstrem
+ * @throws Will throw an error if it can't find an upstream remote.
+ */
+function findUpstream() {
+  console.log('>>> Finding the upstream remote.');
+  try {
+    const upstream = logAndPipeExec(
+      ['git', 'remote', '-v'],
+      ['grep', '-E', 'canova/perf.html|firefox-devtools/profiler'], // TODO: change 'canova' after
+      ['head', '-1'],
+      ['awk', '{ print $1 }']
+    ).trim();
+
+    console.log(`Found '${upstream}' as upstream remote.`);
+    return upstream;
+  } catch (error) {
+    throw new Error(stripIndent`
+      Couldn't find the upstream remote. Is it well configured?
+      We're looking for either devtools-html/perf.html or firefox-devtools/profiler.
+    `);
+  }
+}
+
+/**
+ * Compares the `compareBranch` with `baseBranch` and checks the changed files.
+ * Fails if the `compareBranch` has changes from the files that doesn't match
+ * the `matchRegexp`.
+ *
+ * @param {string} upstream Name of the upstream remote.
+ * @param {string} compareBranch Branch that we want to compare.
+ * @param {string} baseBranch Branch that we want to take as base for comparison.
+ * @param {RegExp} matchRegexp Regexp that will be used to match the files.
+ * @throws Will throw an error if `compareBranch` has changes from the files
+ * that doesn't match the `matchRegexp`.
+ */
+async function checkAllowedPaths(
+  upstream,
+  compareBranch,
+  baseBranch,
+  matchRegexp
+) {
```

Comment:
It would be good to have an option object instead of separate parameters, so that at the call site baseBranch / compareBranch are clearer.